src/main/java/org/akip/service/ProcessTimelineDefinitionService.java

```diff
@@ -0,0 +1,121 @@
+import org.akip.service.dto.TaskInstanceDTO;
+import org.akip.service.mapper.ProcessTimelineDefinitionMapper;
+import org.camunda.bpm.model.bpmn.Bpmn;
+import org.camunda.bpm.model.bpmn.BpmnModelInstance;
+import org.camunda.bpm.model.bpmn.instance.UserTask;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.io.ByteArrayInputStream;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class ProcessTimelineDefinitionService {
+
+    @Autowired
+    private ProcessDefinitionRepository processDefinitionRepository;
+
+    @Autowired
+    private ProcessDeploymentRepository processDeploymentRepository;
+
+    @Autowired
+    private ProcessTimelineDefinitionRepository processTimelineDefinitionRepository;
+
+    private final ProcessTimelineDefinitionMapper processTimelineDefinitionMapper;
+
+    public ProcessTimelineDefinitionService(ProcessTimelineDefinitionMapper processTimelineDefinitionMapper) {
+        this.processTimelineDefinitionMapper = processTimelineDefinitionMapper;
+    }
+
+    public List<ProcessTimelineDefinitionDTO> getByBpmnProcessDefinitionId(String bpmnProcessDefinitionId) {
+        List<ProcessTimelineDefinitionDTO> processTimelineDefinitions = findAllByProcessDefinitionBpmnProcessDefinitionId(
+            bpmnProcessDefinitionId
+        );
+
+        if (processTimelineDefinitions.isEmpty()) {
+            throw new BadRequestErrorException("Timeline not defined for the process");
+        }
+
+        return processTimelineDefinitions;
+    }
+
+    public ProcessTimelineDefinitionDTO save(
+        String processDefinitionBusinessKey,
+        ProcessTimelineDefinitionDTO processTimelineDefinitionDTO
+    ) {
+        AtomicInteger stepCount = new AtomicInteger();
+        Optional<ProcessDefinition> processDefinition = processDefinitionRepository.findByBpmnProcessDefinitionId(
+            processDefinitionBusinessKey
+        );
+        ProcessTimelineDefinition processTimelineDefinition = processTimelineDefinitionMapper.toEntity(processTimelineDefinitionDTO);
+        processTimelineDefinition
+            .getItems()
+            .forEach(
+                item -> {
+                    item.setProcessTimelineDefinition(processTimelineDefinition);
+                    item.setStep(stepCount.getAndIncrement());
+                }
+            );
+        processTimelineDefinition.setProcessDefinition(processDefinition.get());
+        return processTimelineDefinitionMapper.toDto(processTimelineDefinitionRepository.save(processTimelineDefinition));
+    }
+
+    @Transactional(readOnly = true)
+    public List<ProcessTimelineDefinitionDTO> findAllByProcessDefinitionBpmnProcessDefinitionId(String processDefinitionBusinessKey) {
```

Comment:
You should remove this method (see comment above), but what is this property `processDefinitionBusinessKey`?